perllib/FixMyStreet/App/Controller/Report.pm

```diff
@@ -377,15 +377,20 @@ sub inspect : Private {
             if ( $problem->state eq 'hidden' ) {
                 $problem->get_photoset->delete_cached;
             }
-            if ( $problem->state eq 'duplicate' && $old_state ne 'duplicate' ) {
-                # If the report is being closed as duplicate, make sure the
-                # update records this.
-                $update_params{problem_state} = "duplicate";
-            }
-            if ( $problem->state ne 'duplicate' ) {
+            if ( $problem->state eq 'duplicate') {
+                if (my $duplicate_of = $c->get_param('duplicate_of')) {
+                    $problem->set_duplicate_of($duplicate_of);
+                    if ( $old_state ne 'duplicate' ) {
+                        # If the report is being closed as duplicate, make sure the
```

Comment:
Is this now not covered by line 399, which always sets it.

Reply:
👍